aten/src/ATen/native/quantized/cpu/fake_quantize_per_tensor_affine.cpp

```diff
@@ -21,49 +24,31 @@ namespace {
 Notes:
   - quant_delay might be set to non-zero to help weights stabilize in the
     beginning of the training.
-  - quantization range [0, 2^bits - 1]
+  - quantization range [quant_min, quant_max]
 */
-
-class FakeQuantizePerTensorAffineOp_forward : public c10::OperatorKernel {
- public:
-  at::Tensor operator()(
-      at::Tensor X,
+Tensor fake_quantize_per_tensor_affine_cpu(
+      const Tensor& self,
       double scale,
       int64_t zero_point,
-      int64_t quant_min = 0,
-      int64_t quant_max = 255,
-      int64_t quant_delay = 0,
-      int64_t iter = 0
+      int64_t quant_min,
+      int64_t quant_max
     ) {
-    // Sanity checks.
-    if (quant_min > quant_max) {
-      throw std::invalid_argument("`quant_min` should be less than or equal to `quant_max`.");
-    }
-    if (zero_point < 0) {
-      throw std::invalid_argument("`zero_point` must be a positive integer.");
-    }
-    if (quant_delay < 0) {
-      throw std::invalid_argument("`quant_delay` must be a positive integer.");
-    }
-
-    if (quant_delay != 0 && iter < 0) {
-      throw std::invalid_argument(
-        "`iter` must be >=0 for non-zero `quant_delay`");
-    }
-
-    auto Y = at::empty_like(X);
-
-    if (quant_delay > 0 && iter <= quant_delay) {
-      Y.copy_(X);  // We might want to just return the input here.
-      return Y;
-    }
+    TORCH_CHECK(self.scalar_type() == ScalarType::Float);
```

Comment:
in a sense of generality - anything preventing you supporting other float types?

Reply:
we only implemented quantize_linear and dequantize for float right now, but we could also do the double to float conversion when we convert to quantized model. Not sure how valuable it is to support more types though, do we train in double in practice?